MapleServer2/Types/GlobalEvent.cs

```diff
@@ -1,42 +1,60 @@
-﻿using MapleServer2.Packets;
+﻿using MapleServer2.Data.Static;
+using MapleServer2.Packets;
 using MapleServer2.Servers.Game;
 using MapleServer2.Tools;
 
 namespace MapleServer2.Types;
 
 public class GlobalEvent
 {
-    public readonly int Id;
+    public int Id;
+    public int FirstHour;
+    public int FirstMinutesOnHour;
+    public int MinutesToRunPerDay;
     public List<GlobalEventType> Events = new();
 
     public GlobalEvent()
     {
         Id = GuidGenerator.Int();
     }
 
-    public async Task Start()
+    public void Start()
     {
-        MapleServer.BroadcastPacketAll(GlobalPortalPacket.Notice(this));
+        // Assign new ID
+        Id = GuidGenerator.Int();
+        GameServer.GlobalEventManager.AddEvent(this);
+        BroadcastEvent();
+    }
+
+    public async Task BroadcastEvent()
+    {
+        List<Player> onlinePlayers = GameServer.PlayerManager.GetAllPlayers().Where(x => !MapMetadataStorage.MapIsInstancedOnly(x.MapId)).ToList();
+        foreach (Player player in onlinePlayers)
+        {
+            player.Session?.Send(GlobalPortalPacket.Notice(this));
+        }
 
         await Task.Delay(60000);
 
         MapleServer.BroadcastPacketAll(GlobalPortalPacket.Clear(this));
         GameServer.GlobalEventManager.RemoveEvent(this);
     }
 }
+
 public enum GlobalEventType : byte
 {
-    oxquiz = 1,
-    trap_master = 2,
-    spring_beach = 3,
-    crazy_runner = 4,
-    final_surviver = 5,
-    great_escape = 6,
-    dancedance_stop = 7,
-    crazy_runner_shanghai = 8,
-    hideandseek = 9,
-    red_arena = 10,
-    blood_mine = 11,
-    treasure_island = 12,
-    christmas_dancedance_stop = 13
+    none = 0,
+    s_massive_event_name_oxquiz = 1,
+    s_massive_event_name_trap_master = 2,
+    s_massive_event_name_spring_beach = 3,
+    s_massive_event_name_crazy_runner = 4,
+    s_massive_event_name_final_surviver = 5,
+    s_massive_event_name_great_escape = 6,
+    s_massive_event_name_dancedance_stop = 7,
+    s_massive_event_name_crazy_runner_shanghai = 8,
+    s_massive_event_name_hideandseek = 9,
+    s_massive_event_name_red_arena = 10,
+    s_massive_event_name_blood_mine = 11,
+    s_massive_event_name_treasure_island = 12,
+    s_massive_event_name_christmas_dancedance_stop = 13
```

Comment:
IMO this was better before - the Enum values are harder to read now, and harder to find the value you want via intellisense.

Reply:
I think these names are from xml, I'm correct? @Zintixx

Reply:
They are but I can revert to what it was before, to regain that readability (removing the "s_massive_event_name_" part of the name and doing something like this:
![image](https://user-images.githubusercontent.com/62830952/156905640-9cdb132a-ca46-4f00-82fc-3056889d6f47.png)

Reply:
What if you revert the naming and just create a function that inputs the enum and outputs the string `"s_massive_event_name_" + eventType`?